### PR TITLE
Remove this.resource from RouterDSL

### PR DESCRIPTION
**Pull request by locks**

### CODE REVIEW

**Comment by rwjblue** [comment]
Since this is exposed as `Ember.RouterDSL` I believe we can definitely move the implementation directly into the legacy addon. Just need to make an issue on ember-2-legacy for this one, and cross link back here.

File: packages/ember-routing/tests/system/dsl_test.js [hunk 1]
```diff
@@ -18,7 +18,6 @@ QUnit.module('Ember Router DSL', {
 });
 
 QUnit.test('should fail when using a reserved route name', function() {
-  expectDeprecation('this.resource() is deprecated. Use this.route(\'name\', { resetNamespace: true }, function () {}) instead.');
   let reservedNames = ['array', 'basic', 'object', 'application'];
 
   expect((reservedNames.length * 2) + 1);
```
Comment (rwjblue) [inline packages/ember-routing/tests/system/dsl_test.js]: Need to tweak this somewhat, the CI failure is due to it thinking it has 9 assertions and only actually getting 5

**Comment by rwjblue** [comment]
@locks - Needs a rebase...

**Comment by locks** [comment]
@rwjblue rebased

**Review by rwjblue — APPROVED** [review]

**Pull request merged**